src/main/java/se/liu/ida/hefquin/queryplan/executable/impl/ops/ExecOpGenericBindJoin.java

```diff
@@ -0,0 +1,78 @@
+package se.liu.ida.hefquin.queryplan.executable.impl.ops;
+
+import se.liu.ida.hefquin.data.SolutionMapping;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMapping;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMappingUtils;
+import se.liu.ida.hefquin.federation.FederationMember;
+import se.liu.ida.hefquin.query.Query;
+import se.liu.ida.hefquin.queryplan.executable.IntermediateResultBlock;
+import se.liu.ida.hefquin.queryplan.executable.IntermediateResultElementSink;
+import se.liu.ida.hefquin.queryproc.ExecutionContext;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+public abstract class ExecOpGenericBindJoin<QueryType extends Query, MemberType extends FederationMember>
+        implements UnaryExecutableOp
+{
+    protected final QueryType query;
+    protected final MemberType fm;
+
+    public ExecOpGenericBindJoin(final QueryType query, final MemberType fm ) {
+        assert query != null;
+        assert fm != null;
+
+        this.query = query;
+        this.fm = fm;
+    }
+
+    @Override
+    public int preferredInputBlockSize() {
+        // Since this algorithm processes the input solution mappings
+        // sequentially (one at a time), and input block size of 1 may
+        // reduce the response time of the overall execution process.
+        return 1;
```

Comment:
This does not make sense for bind joins.

Reply:
I have updated the preferredInputBlockSize for bind join (See commit db1ce88). 
According to the experiment result of the BrTPF paper, I choose 30 as the initial block size for bind join, but it can be optimized in query planning. Please let me know if it makes sense.